src/common/selftest/selftest_axis.h

```diff
@@ -0,0 +1,51 @@
+// selftest_axis.h
+#pragma once
+
+#include <inttypes.h>
+#include "selftest_MINI.h"
+
+typedef struct _selftest_axis_config_t {
+    const char *partname;
+    uint8_t axis;
+} selftest_axis_config_t;
+
+class CSelftestPart_Axis : public CSelftestPart {
+public:
+    enum TestState : uint8_t {
+        spsIdle,
+        spsStart,
+        spsWaitHome,
+        spsFinish,
+        spsFinished,
+        spsAborted,
+    };
+    enum HomeState : uint8_t {
+        hsNone,
+        hsHommingInProgress,
+        hsHommingFinished,
+    };
+
+public:
+    CSelftestPart_Axis(const selftest_axis_config_t *pconfig);
+
+public:
+    bool IsInProgress() const;
+
+public:
+    bool Start();
```

Comment:
parent has this method virtual, so this one is virtual too. Use virtual and override so it is obvious
virtual  bool Start() override;

Reply:
fixed